site/run-letsencrypt.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -ex
+
+# start service
+kubectl apply -f service.yaml
+
+# stop existing site deployment
+kubectl delete --ignore-not-found=true -f site-deployment.yaml
+N=
+while [[ $N != 0 ]]; do
+    sleep 5
+    N=$(kubectl get pods -l app=site --ignore-not-found=true --no-headers | wc -l | tr -d '[:space:]')
+    echo N=$N
+done
+
+# stop existing run-letsencrypt pod
+kubectl delete pod --ignore-not-found=true run-letsencrypt
+N=
+while [[ $N != 0 ]]; do
+    sleep 5
+    N=$(kubectl get pod --ignore-not-found=true --no-headers run-letsencrypt | wc -l | tr -d '[:space:]')
+    echo N=$N
+done
+
+# run run-letsencrypt pod
+kubectl create -f run-letsencrypt-pod.yaml
+echo "Waiting for run-letsencrypt to complete..."
+EC=""
+while [[ $EC = "" ]]; do
+    sleep 5
+    EC=$(kubectl get pod -o "jsonpath={.status.containerStatuses[0].state.terminated.exitCode}" run-letsencrypt)
```

Comment:
will `run-letsencrypt` always eventually succeed or time-out, setting EC to non-empty string?

Reply:
Just to explain this, certbot is the Let's Encrypt tool for getting TLS certificates.  You have to run "certbot run" once to get the initial certs.  Then, a job, `certbot renew` runs periodically (normally daily, run by cron) to see if those certs need to be renewed.

I suppose it could go into an infinite loop, although I think certbot (run inside of run-letsencrypt) has a timeout.  This is currently run by hand.  We might want to do more if we automate it as part of building the cluster, although failure means: the configuration is bad, in which case should should fail, or Let's Encrypt is down, in which case I guess you want to retry if certbot doesn't.